lib/index.ts

```diff
@@ -0,0 +1,296 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2021 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { randomBytes } from 'crypto';
+
+import { Transformer } from './transform';
+import { Engine } from './engine';
+import { convertDistance } from './util/convertDistance';
+import { convertGeopoint } from './util/convertGeopoint';
+import { hash } from './util/hash';
+import { JSONObject } from './types/JSONObject';
+
+
+class NormalizedFilter {
+  public filter: any;
+  public id: string;
+  public index: string|null;
+
+  constructor (normalized: any, id: string, index: string|null) {
+    this.filter = normalized;
+    this.id = id;
+    this.index = index;
+  }
+}
+
+export interface KoncordeOptions {
+  maxMinTerms: number;
+  regExpEngine: string;
+  seed: ArrayBuffer;
+}
```

Comment:
It's considered as a good practice to add inline documentation to the interface so it can be displayed by the IDE

```suggestion
export interface KoncordeOptions {
  /**
   * Example documentation
   */
  maxMinTerms: number;
  regExpEngine: string;
  seed: ArrayBuffer;
}
```